riscv/triggers.cc

```diff
@@ -234,6 +234,62 @@ std::optional<match_result_t> mcontrol_common_t::detect_memory_access_match(proc
   return std::nullopt;
 }
 
+reg_t mcontrol6_t::tdata1_read(const processor_t * const proc) const noexcept {
+  unsigned xlen = proc->get_const_xlen();
+  reg_t tdata1 = 0;
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_TYPE(xlen), 6);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_DMODE(xlen), dmode);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_VS, proc->extension_enabled('H') ? vs : 0);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_VU, proc->extension_enabled('H') ? vu : 0);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_HIT, hit);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_SELECT, select);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_TIMING, timing);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_ACTION, action);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_CHAIN, chain);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_MATCH, match);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_M, m);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_S, s);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_U, u);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_EXECUTE, execute);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_STORE, store);
+  tdata1 = set_field(tdata1, CSR_MCONTROL6_LOAD, load);
+  return tdata1;
+}
+
+void mcontrol6_t::tdata1_write(processor_t * const proc, const reg_t val, const bool allow_chain) noexcept {
+  auto xlen = proc->get_const_xlen();
+  assert(get_field(val, CSR_MCONTROL6_TYPE(xlen)) == CSR_TDATA1_TYPE_MCONTROL6);
+  dmode = get_field(val, CSR_MCONTROL6_DMODE(xlen));
+  vs = get_field(val, CSR_MCONTROL6_VS);
+  vu = get_field(val, CSR_MCONTROL6_VU);
+  hit = get_field(val, CSR_MCONTROL6_HIT);
+  select = get_field(val, CSR_MCONTROL6_SELECT);
+  timing = get_field(val, CSR_MCONTROL6_TIMING);
+  action = legalize_action(get_field(val, CSR_MCONTROL6_ACTION));
+  chain = allow_chain ? get_field(val, MCONTROL_CHAIN) : 0;
+  unsigned match_value = get_field(val, CSR_MCONTROL6_MATCH);
+  switch (match_value) {
+    case MATCH_EQUAL:
+    case MATCH_NAPOT:
+    case MATCH_GE:
+    case MATCH_LT:
+    case MATCH_MASK_LOW:
+    case MATCH_MASK_HIGH:
+      match = (match_t)match_value;
+      break;
+    default:
+      match = MATCH_EQUAL;
+  }
```

Comment:
Create a `legalize_match()` in `mcontrol_common_t` instead of copy/pasting this paragraph from `mcontrol_t`